packages/react-components/src/components/Switch/Switch.tsx

```diff
@@ -42,9 +49,41 @@ export const Switch: React.FC<SwitchProps> = ({
     }
   }, [on]);
 
-  const iconSize: IconSize = size === 'basic' ? 'small' : 'xsmall';
+  const getIcon = (iconSize: IconSize): JSX.Element => {
+    if (loading) {
+      return <Loader size="cover" />;
+    }
+
+    if (disabled) {
+      return (
+        <Icon
+          data-testid={`${baseClass}__disabled-icon`}
+          size={iconSize}
+          source={LockIcon}
+          kind={'primary'}
+        />
+      );
+    }
+
+    if (on) {
+      return (
+        <Icon
+          className={styles[`${baseClass}__icon--on`]}
+          size={iconSize}
+          source={CheckIcon}
+          kind={'primary'}
+        />
+      );
+    }
+
+    return <Icon size={iconSize} source={CloseIcon} kind={'primary'} />;
```

Comment:
Why this one does not require a classname? I think you can handle them without separating them by the if clause. Just make sure the props provided are defined conditionally (icon and className)

Reply:
☝️ it's also removed (there won't be any icon for `off` state)